compiler/haskell-ide-core/src/Development/IDE/Functions/CPP.hs

```diff
@@ -52,11 +53,12 @@ doCpp dflags raw input_fn output_fn = do
                       | otherwise = SysTools.runCc
                                           dflags (SysTools.Option "-E" : args)
 
-    let target_defs = [] {-
-          [ "-D" ++ HOST_OS     ++ "_BUILD_OS",
-            "-D" ++ HOST_ARCH   ++ "_BUILD_ARCH",
-            "-D" ++ TARGET_OS   ++ "_HOST_OS",
-            "-D" ++ TARGET_ARCH ++ "_HOST_ARCH" ] -}
+    let target_defs =
+          -- NEIL: Patched to use System.Info instead of constants from CPP
```

Comment:
What exactly broke the existing code?

Reply:
Before I had commented out all these defines, when hacking over the CPP module in the first place. Shake has some `#ifdef mingw32_HOST_OS` and that wasn't defined, despite me being on Windows.

Reply:
Oooh, I missed the block comment around this. Now this makes a lot more sense :)